react-native/architecture/components/my-component/component.s.md

```diff
@@ -0,0 +1,16 @@
+# [Standard] Components folder - Component file
+
+## Standard
+
+- The file name should be `MyComponent.style.js`
+- Exports the component class by default
+
+  > **Why?:** Linked to code splitting and single responsability. Facilitates code testing.
+
+- Contains a unique `render()`
+
+  > **Why?:** Multiple renders imply multiple components
```

Comment:
How to deal with FlatList like renderItems:

Is this correct ?
```js
class MyList extends React.PureComponent {
  state = {selected: (new Map(): Map<string, boolean>)};

  _keyExtractor = (item, index) => item.id;

  _onPressItem = (id: string) => {
    // updater functions are preferred for transactional updates
    this.setState((state) => {
      // copy the map rather than modifying state.
      const selected = new Map(state.selected);
      selected.set(id, !selected.get(id)); // toggle
      return {selected};
    });
  };

  _renderItem = ({item}) => (
    <MyListItem
      id={item.id}
      onPressItem={this._onPressItem}
      selected={!!this.state.selected.get(item.id)}
      title={item.title}
    />
  );

  render() {
    return (
      <FlatList
        data={this.props.data}
        extraData={this.state}
        keyExtractor={this._keyExtractor}
        renderItem={this._renderItem}
      />
    );
  }
}
```

Reply:
Je suis plutôt ok sur le nombre de lignes max. Après il faut trouver une définition de ce qu'on appelle un "render" dans cette discussion car trop souvent j'ai vu l'implémentation de `MyListItem` dans `_renderItem` plutôt qu'une simple instanciation du composant comme dans l'exemple de Tycho